shared/src/pubsub/subscriber.rs

```diff
@@ -43,4 +44,48 @@ impl Subscriber {
             },
         )))
     }
+
+    pub async fn subscribe_alt<M: MessagePayload>(
+        &self,
+    ) -> Result<Receiver<std::pin::Pin<Box<dyn Stream<Item = Envelope<M>> + Send>>>, SubscriberError>
```

Comment:
the return type should be the same as before. The channel shouldn't transmit the stream but the items.